docs/command_reference.md

```diff
@@ -0,0 +1,299 @@
+
+## Initializing a trusted collection
+
+Notary can initialize a trusted collection with the `notary init` command:
+```bash
+$ notary init <GUN>
+```
+
+This command will generate targets and snapshot keys locally for the trusted collection, and try to locate a root key to use in the specified notary client cache (from `-d` or the config).
+If notary cannot find a root key, it will generate one.  For all keys, notary will also prompt for a passphrase to encrypt the private key material at rest.
+
+If you'd like to initialize your trusted collection with a specific root key, there is a flag to provide it.  Notary will require that the key is encrypted:
+```bash
+$ notary init <GUN> --rootkey <key_file>
+```
+
+Note that you will have to run a publish after this command for it to take effect, as there are staged changes to initialize the trusted collection that have not been pushed to a notary server.
+```bash
+$ notary publish <GUN>
+```
+
+## Manage staged changes
+
+The Notary CLI client stages changes before publishing them to the server.
+These changes are staged locally in files such that the client can request the latest updates from the notary server, attempt to apply the staged changes on top of the new updates (like a `git rebase`), and then finally publish if the changes are still valid.
+
+You can manage the changes that are staged by running:
+
+```bash
+# Check what changes are staged
+$ notary status <GUN>
+
+# Unstage a specific change
+$ notary status <GUN> --unstage 0
+
+# Alternatively, unstage all changes
+$ notary status <GUN> --reset
+```
+
+Note that `<GUN>` can take on arbitrary values, but when working with Docker Content Trust they are structured like `<domain>/<account>/<repository>`.  For example `docker.io/library/alpine` for the [Docker Library Alpine image](https://hub.docker.com/r/library/alpine).
+
+When you're ready to publish your changes to the Notary server, run:
+
+```bash
+$ notary publish <GUN>
+```
+
+## Auto-publish changes
+
+Instead of manually running `notary publish` after each command, you can use the `-p` flag to auto-publish the changes from that command.
+For example:
+
+```bash
+$ notary init -p <GUN>
+```
+
+The remainder of this reference will include the `-p` auto-publish flag where it can be used, though it is optional and can be replaced with following each command with a `notary publish`.
+
+## Adding and removing trust data
+
+Users can sign content into a notary trusted collection by running:
+```bash
+$ notary add -p <GUN> <target_name> <target_file>
+```
+
+In the above command, the `<target_name>` corresponds to the name we want to associate the `<target_file>`'s signature with. Notary will sign the hash of the `<target_file>` into its trusted collection.
```

Comment:
Non-blocking:  should we use the word "signature" here, since the rest of notary is about signing things?  Since we're not specifically signing this `<target_file>` itself, maybe something like "`<target_name>` corresponds to the name we want `<target_file>` to be known as in the trusted collection"?